src/reactive/infoblox_handlers.py

```diff
@@ -24,44 +30,42 @@
 use_defaults('update-status')
 
 
-@reactive.when_not('infoblox.installed')
+@when_not('infoblox.installed')
+@when('endpoint.neutron.connected')
 def install_infoblox():
     with provide_charm_instance() as charm_class:
         charm_class.install()
-    reactive.set_state('infoblox.installed')
-
-
-@reactive.hook('config-changed')
-def config_changed():
-    reactive.remove_state('neutron.configured')
-    reactive.remove_state('designate.configured')
+    set_flag('infoblox.installed')
 
 
-@reactive.when('infoblox.create-defs')
-@reactive.when('infoblox.installed')
-@reactive.when_not('infoblox.ready')
+@when_all('endpoint.neutron.neutron_server_ready',
+          'infoblox.installed')
+@when_not('infoblox.ready')
 def create_ea_definitions():
     with provide_charm_instance() as charm_class:
         charm_class.create_ea_definitions()
-    reactive.set_state('infoblox.ready')
+    status_set('active', 'Unit is ready')
+    set_flag('infoblox.ready')
 
 
-@reactive.when('neutron.connected')
-@reactive.when('infoblox.installed')
-@reactive.when_not('neutron.configured')
+@when_all('infoblox.installed',
+          'endpoint.neutron.connected')
+@when_not('endpoint.neutron.configured')
 def configure_neutron(principle):
     with provide_charm_instance() as charm_class:
         config = charm_class.get_neutron_conf()
-        principle.configure_principal(config)
-    reactive.set_state('neutron.configured')
+        principal_neutron = \
+            endpoint_from_flag('endpoint.neutron.connected')
+        principal_neutron.configure_principal(config)
+    set_flag('endpoint.neutron.configured')
```

Comment:
This is a no-no.  The only thing that should set endpoint flags is an Endpoint class in the the relevant interface.  This should either be a general flag in the charm/layer or a function call into the class.  e.g. `principal_neutron.is_now_configured()`  -- however, that's not a great design pattern, as Interfaces/Endpoints should be thin and dumb.

Reply:
Based on our discussion, I changed the infoblox interface.